### DISCUSSION TITLE
How to silent install without it running using winget?

### DISCUSSION THREAD

**Discussion by eggbean** [Devolutions/UniGetUI#2314] [Q&A] [answered]
I have a script which installs stuff on a new installation of Windows, including WingetUI. It runs as admin, as it does a lot of other stuff as well.

    winget install -e --id SomePythonThings.WingetUIStore

This makes WingetUI run after installation and that also makes other things not install afterwards for some reason.

    winget install -e --id SomePythonThings.WingetUIStore --override "/NoAutoStart"

Now I get asked if I want to install for all users or just the current user.

    winget install -e --id SomePythonThings.WingetUIStore --override "/NoAutoStart /ALLUSERS"

Now I get asked which language.

    winget install -e --id SomePythonThings.WingetUIStore --override "/NoAutoStart /ALLUSERS /LANG=english"

Now the installer dialog thing comes up.

    winget install -e --id SomePythonThings.WingetUIStore --override "/SP /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /NoAutoStart /ALLUSERS /LANG=english"

How the program runs after installation (as admin) and things are no better than they were when I started.

How do I do this?

**Comment by marticliment** [comment]
What do you want to achieve?

**Comment by eggbean** [comment]
I want to install WingetUI using winget within a setup script and not have WingetUI automatically start after it's installed and requiring no user input (unattended).

https://superuser.com/questions/1845737/packages-being-skipped-when-installing-using-winget-sequentially

**Reply by marticliment** [reply, marked as answer]
My excuses, it seems the flag `/NoAutoStart` broke somehow. I have fixed this issue, and the next installer should have this issue fixed.

The last command-line you shared should work then

**Comment by eggbean** [comment]
Thanks. Just to confirm, will all of these really be required?

`/SP /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /NoAutoStart /ALLUSERS /LANG=english`

**Reply by marticliment** [reply]
Yes, since you are overriding the options in which WinGet calls the executable

**Comment by eggbean** [comment]
Now that your fix is in the current version, `/NoAutoStart` is working now so the application doesn't start after installation.

However, it always reinstalls when I run the script, whereas all other packages are only reinstalled if there is an new version compared to the one installed. It's not a big deal, but why is it happening? This is what I am using:

    winget install -e --id SomePythonThings.WingetUIStore --override "/SP /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /NoAutoStart /ALLUSERS /LANG=english"

```
Found an existing package already installed. Trying to upgrade the installed package...
No available upgrade found.
No newer package versions are available from the configured sources.
Found UniGetUI (formerly WingetUI) [SomePythonThings.WingetUIStore] Version 3.1.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Successfully verified installer hash
Starting package install...
Found an existing package already installed. Trying to upgrade the installed package...
No available upgrade found.
No newer package versions are available from the configured sources.
Found an existing package already installed. Trying to upgrade the installed package...
No available upgrade found.

```

**Reply by marticliment** [reply]
I do not know, I will investigate

**Reply by eggbean** [reply]
Thanks.
